com.avaloq.tools.ddk.check.runtime.ui/src/com/avaloq/tools/ddk/check/runtime/ui/validation/CheckMarkerUpdateJob.java

```diff
@@ -209,43 +209,72 @@ protected IStatus run(final IProgressMonitor monitor) {
+  private void validateAndCreateMarkers(final IResourceValidator resourceValidator, final MarkerCreator markerCreator, final IFile file, final Resource resource, final IProgressMonitor monitor) {
     try {
       monitor.subTask("validating " + file.getName()); //$NON-NLS-1$
-
-      final List<Issue> list = resourceValidator.validate(resource, checkMode, getCancelIndicator(monitor));
-      if (list != null) {
-        // resourceValidator.validate returns null if canceled (and not an empty list)
-        file.deleteMarkers(MarkerTypes.FAST_VALIDATION, true, IResource.DEPTH_ZERO);
-        file.deleteMarkers(MarkerTypes.NORMAL_VALIDATION, true, IResource.DEPTH_ZERO);
-        file.deleteMarkers(MarkerTypes.EXPENSIVE_VALIDATION, true, IResource.DEPTH_ZERO);
-
-        if (markerCreator != null) {
-          for (final Issue issue : list) {
-            markerCreator.createMarker(issue, file, MarkerTypes.forCheckType(issue.getType()));
-          }
-        } else {
-          if (LOGGER.isDebugEnabled()) {
-            LOGGER.error("Could not create markers. The marker creator is null."); //$NON-NLS-1$
-          }
-        }
-      }
+      final Collection<Issue> issues = validateResource(resourceValidator, resource, monitor);
+      createMarkers(markerCreator, file, issues);
     } catch (final CoreException e) {
       LOGGER.error(e.getMessage(), e);
     } finally {
       monitor.worked(1);
     }
   }
 
+  /**
+   * Creates the marker on a given file based on the list of issues.
+   *
+   * @param markerCreator
+   *          the marker creator
+   * @param file
+   *          the EFS file (not null)
+   * @param issues
+   *          the list of issues
+   * @throws CoreException
+   */
+  private void createMarkers(final MarkerCreator markerCreator, final IFile file, final Collection<Issue> issues) throws CoreException {
+    if (issues != null) {
+      // resourceValidator.validate returns null if canceled (and not an empty list)
+      file.deleteMarkers(MarkerTypes.FAST_VALIDATION, true, IResource.DEPTH_ZERO);
+      file.deleteMarkers(MarkerTypes.NORMAL_VALIDATION, true, IResource.DEPTH_ZERO);
+      file.deleteMarkers(MarkerTypes.EXPENSIVE_VALIDATION, true, IResource.DEPTH_ZERO);
+      if (markerCreator != null) {
+        for (final Issue issue : issues) {
+          markerCreator.createMarker(issue, file, MarkerTypes.forCheckType(issue.getType()));
+        }
+      } else {
+        if (LOGGER.isDebugEnabled()) {
+          LOGGER.error("Could not create markers. The marker creator is null."); //$NON-NLS-1$
+        }
+      }
+    }
+  }
+
+  /**
+   * Validates the given resource and returns a list of {@link Issue} based on validation.
+   *
+   * @param resourceValidator
+   *          the resource validator
+   * @param resource
+   *          the EMF resource (not null)
+   * @param monitor
+   *          the monitor (not null)
+   * @return the list of {@link Issue} (not null)

```

Comment:
... contradict this. Besides, it returns collection, so let's say "collection" instead of "list."

Reply:
Also, the current coding convention seems to be `@returns bla bla bla, never {@code null}` or `@param bla bla bla, must not be/may be {@code null}`. It would be nice to update this class to match.

Reply:
ok. done